apps/game_backend/lib/game_backend/users.ex

```diff
@@ -388,6 +389,10 @@ defmodule GameBackend.Users do
         Currencies.substract_currencies(user_id, upgrade.cost)
       end)
       |> Transaction.run()
+      |> case do
+        {:ok, _} -> {:ok, get_user(user_id)}
+        {:error, _} -> {:error, :unknown_error}
```

Comment:
I think this case is wrong right? multi error result is a cuadruple tuple {:error, key, value, changes so far}